From a2f846a1fd7ac8b5225114ebf5b50c53dffcce52 Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Thu, 28 Apr 2016 12:19:18 +0800
Subject: [PATCH 1192/5242] MLK-12928-8 ARM: dts: imx6ul-14x14-evk: add new
 imx6ul-14x14-evk-pf1550 boad

commit  50ea437808a032a22d46b74dd40be27a3090f50f from
https://source.codeaurora.org/external/imx/linux-imx.git

Add new pf1550 board support.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/Makefile                    |    1 +
 arch/arm/boot/dts/imx6ul-14x14-evk-pf1550.dts |  115 +++++++++++++++++++++++++
 arch/arm/boot/dts/imx6ul-14x14-evk.dtsi       |    8 +-
 3 files changed, 123 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm/boot/dts/imx6ul-14x14-evk-pf1550.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index ff6b8f3..a92fb53 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -597,6 +597,7 @@ dtb-$(CONFIG_SOC_IMX6UL) += \
 	imx6ul-14x14-ddr3-arm2-spdif.dtb \
 	imx6ul-14x14-ddr3-arm2-wm8958.dtb \
 	imx6ul-14x14-lpddr2-arm2.dtb \
+	imx6ul-14x14-evk-pf1550.dtb \
 	imx6ul-9x9-evk.dtb \
 	imx6ul-9x9-evk-csi.dtb \
 	imx6ul-9x9-evk-btwifi.dtb \
diff --git a/arch/arm/boot/dts/imx6ul-14x14-evk-pf1550.dts b/arch/arm/boot/dts/imx6ul-14x14-evk-pf1550.dts
new file mode 100644
index 0000000..5c60b1a
--- /dev/null
+++ b/arch/arm/boot/dts/imx6ul-14x14-evk-pf1550.dts
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2016 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6ul-14x14-evk.dts"
+
+&cpu0 {
+	/*
+	 * on i.MX6UL, no separated VDD_ARM_IN and VDD_SOC_IN,
+	 * to align with other platform and use the same cpufreq
+	 * driver, still use the separated OPP define for arm
+	 * and soc.
+	 */
+	operating-points = <
+		/* kHz	uV */
+		696000  1275000
+		528000	1175000
+		396000	1175000
+		198000	1175000
+	>;
+	fsl,soc-operating-points = <
+		/* KHz	uV */
+		696000  1275000
+		528000	1175000
+		396000	1175000
+		198000	1175000
+	>;
+	arm-supply = <&sw1_reg>;
+	soc-supply = <&sw1_reg>;
+	fsl,arm-soc-shared = <1>;
+};
+
+&gpc {
+	fsl,ldo-bypass = <1>; /* use ldo-bypass */
+};
+
+&i2c1 {
+	pmic: pf1550@08 {
+		compatible = "fsl,pf1550";
+		interrupt-parent = <&gpio5>;
+		interrupts = <4 8>;
+		reg = <0x08>;
+		pinctrl-0 = <&pinctrl_pf1550>;
+
+		regulators {
+			compatible = "pf1550-regulator";
+
+			sw1_reg: SW1 {
+				regulator-name = "SW1";
+				regulator-min-microvolt = <600000>;
+				regulator-max-microvolt = <1387500>;
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-ramp-delay = <6250>;
+			};
+
+			sw2_reg: SW2 {
+				regulator-name = "SW2";
+				regulator-min-microvolt = <600000>;
+				regulator-max-microvolt = <1387500>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			sw3_reg: SW3 {
+				regulator-name = "SW3";
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			vref_reg: VREFDDR {
+				regulator-name = "VREFDDR";
+				regulator-min-microvolt = <1200000>;
+				regulator-max-microvolt = <1200000>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			vldo1_reg: LDO1 {
+				regulator-name = "LDO1";
+				regulator-min-microvolt = <750000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vldo2_reg: LDO2 {
+				regulator-name = "LDO2";
+				regulator-min-microvolt = <1800000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+
+			vldo3_reg: LDO3 {
+				regulator-name = "LDO3";
+				regulator-min-microvolt = <750000>;
+				regulator-max-microvolt = <3300000>;
+				regulator-always-on;
+			};
+		};
+	};
+};
+
+&sai2 {
+	status = "disabled";
+};
+
+&sound {
+	status = "disabled";
+};
diff --git a/arch/arm/boot/dts/imx6ul-14x14-evk.dtsi b/arch/arm/boot/dts/imx6ul-14x14-evk.dtsi
index c09506a..dbbe14c 100644
--- a/arch/arm/boot/dts/imx6ul-14x14-evk.dtsi
+++ b/arch/arm/boot/dts/imx6ul-14x14-evk.dtsi
@@ -71,7 +71,7 @@
 		states = <1300000 0x1 1400000 0x0>;
 	};
 
-	sound {
+	sound: sound {
 		compatible = "simple-audio-card";
 		simple-audio-card,name = "mx6ul-wm8960";
 		simple-audio-card,format = "i2s";
@@ -580,6 +580,12 @@
 		>;
 	};
 
+	pinctrl_pf1550: pf1550 {
+		fsl,pins = <
+			 MX6UL_PAD_SNVS_TAMPER4__GPIO5_IO04      0x80000000
+		>;
+	};
+
 	pinctrl_pwm1: pwm1grp {
 		fsl,pins = <
 			MX6UL_PAD_GPIO1_IO08__PWM1_OUT   0x110b0
-- 
1.7.9.5

